docs/specs/oauth2-with-scopes.yaml

```diff
@@ -0,0 +1,40 @@
+openapi: 3.0.0
```

Comment:
Unfortunately, I was not able to find a suitable public API, so I created this dummy API here.
I tested the functionality, by debugging the network requests and by testing against a real private API. Functionality worked, as expected.